src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/data/ConfiguredApiRepository.kt

```
@@ -29,16 +29,16 @@ class ConfiguredApiRepository {
     pageSize: Long,
     sortColumn: String,
     sortedAsc: Boolean,
+    caseloads: List<String>,
   ): List<Map<String, Any>> {
-    val (preparedStatementNamedParams, whereClause) = buildWhereClause(filtersExcludingRange, rangeFilters)
+    if (caseloads.isEmpty()) {
```

I think this should log a warning, or possibly an error. Also, it's possible that a specific query doesn't need to be filtered by caseload.

Maybe `buildWhereClause` could throw an unauthorised exception if it expected a caseload, but none were provided?

---

I returned an empty list based on the Jira AC that if a user is authenticated and has no caseload they see 0 records and the count returned is 0.
I am not sure if we want to throw an unauthorised exception since the row level security acts as a filter and if nothing matches simply no rows are returned. 
Similarly there could also be another case for example when the user has one or more caseloads and nothing matches and again they see zero results. 
Would we want to treat this also as unauthorised? 
Happy to discuss to clarify this.